var/spack/repos/builtin/packages/libsonata/package.py

```diff
@@ -36,22 +36,24 @@ class Libsonata(CMakePackage):
     homepage = "https://bbpcode.epfl.ch/code/#/admin/projects/common/libsonata"
     url      = "ssh://bbpcode.epfl.ch/common/libsonata"
 
-    version('dev', git=url, submodules=True)
+    version('develop', git=url, submodules=False)
 
     variant('mpi', default=False, description="Enable MPI backend")
+    variant('python', default=False, description="Enable Python bindings")
 
     depends_on('cmake@3.0:', type='build')
-    depends_on('hdf5+mpi', when='+mpi')
-    depends_on('hdf5~mpi', when='~mpi')
+    depends_on('fmt@4.0:')
+    depends_on('highfive+mpi', when='+mpi')
+    depends_on('highfive~mpi', when='~mpi')
     depends_on('mpi', when='+mpi')
-    # TODO: depends_on('python')
-
-    patch('CMakeLists.patch')  # remove '-pedantic' compiler option (it's not happy with MPI headers)
+    depends_on('py-pybind11@2.2', type='build', when='+python')
```

Comment:
This one currently gives me 'Error: maximum recursion depth exceeded' from Spack.
Any hint how can I investigate it?

Reply:
Internally spack has an infinite recursion? In your code I cant see what might be causing it. Could you post some logging?

Reply:
@arsenius7  : I cloned your repo & branch `arsenius7/libsonata` and see:

```
± |arsenius7/libsonata ✓| → spack spec libsonata
Input spec
--------------------------------
libsonata

Concretized
--------------------------------
libsonata@develop%gcc@6.4.0 build_type=RelWithDebInfo ~mpi~python arch=darwin-sierra-x86_64
    ^cmake@3.12.2%gcc@6.4.0~doc+ncurses+openssl+ownlibs patches=dd3a40d4d92f6b2158b87d6fb354c277947c776424aa03f6dc8096cf3135f5d0 ~qt arch=darwin-sierra-x86_64
        ^ncurses@5.9%gcc@6.4.0 patches=f84b2708a42777aadcc7f502a261afe10ca5646a51c1ef8b5e60d2070d926b57 ~symlinks~termlib arch=darwin-sierra-x86_64
        ^openssl@1.0.2k%gcc@6.4.0+systemcerts arch=darwin-sierra-x86_64
    ^fmt@4.1.0%gcc@6.4.0 build_type=RelWithDebInfo arch=darwin-sierra-x86_64
    ^highfive@1.6%gcc@6.4.0~boost build_type=RelWithDebInfo ~mpi arch=darwin-sierra-x86_64
        ^hdf5@1.10.3%gcc@6.4.0~cxx~debug~fortran~hl~mpi+pic+shared~szip~threadsafe arch=darwin-sierra-x86_64
            ^zlib@1.2.11%gcc@6.4.0+optimize+pic+shared arch=darwin-sierra-x86_64


± |arsenius7/libsonata ✓| → spack spec libsonata+mpi
Input spec
--------------------------------
libsonata+mpi

Concretized
--------------------------------
libsonata@develop%gcc@6.4.0 build_type=RelWithDebInfo +mpi~python arch=darwin-sierra-x86_64
    ^cmake@3.12.2%gcc@6.4.0~doc+ncurses+openssl+ownlibs patches=dd3a40d4d92f6b2158b87d6fb354c277947c776424aa03f6dc8096cf3135f5d0 ~qt arch=darwin-sierra-x86_64
        ^ncurses@5.9%gcc@6.4.0 patches=f84b2708a42777aadcc7f502a261afe10ca5646a51c1ef8b5e60d2070d926b57 ~symlinks~termlib arch=darwin-sierra-x86_64
        ^openssl@1.0.2k%gcc@6.4.0+systemcerts arch=darwin-sierra-x86_64
    ^fmt@4.1.0%gcc@6.4.0 build_type=RelWithDebInfo arch=darwin-sierra-x86_64
    ^highfive@1.6%gcc@6.4.0~boost build_type=RelWithDebInfo +mpi arch=darwin-sierra-x86_64
        ^hdf5@1.10.3%gcc@6.4.0~cxx~debug~fortran~hl+mpi+pic+shared~szip~threadsafe arch=darwin-sierra-x86_64
            ^hpe-mpi@2.16%gcc@6.4.0 arch=darwin-sierra-x86_64
            ^zlib@1.2.11%gcc@6.4.0+optimize+pic+shared arch=darwin-sierra-x86_64


± |arsenius7/libsonata ✓| → spack spec libsonata+mpi+python
Input spec
--------------------------------
libsonata+mpi+python

Concretized
--------------------------------
libsonata@develop%gcc@6.4.0 build_type=RelWithDebInfo +mpi+python arch=darwin-sierra-x86_64
    ^cmake@3.12.2%gcc@6.4.0~doc+ncurses+openssl+ownlibs patches=dd3a40d4d92f6b2158b87d6fb354c277947c776424aa03f6dc8096cf3135f5d0 ~qt arch=darwin-sierra-x86_64
        ^ncurses@5.9%gcc@6.4.0 patches=f84b2708a42777aadcc7f502a261afe10ca5646a51c1ef8b5e60d2070d926b57 ~symlinks~termlib arch=darwin-sierra-x86_64
        ^openssl@1.0.2k%gcc@6.4.0+systemcerts arch=darwin-sierra-x86_64
    ^fmt@4.1.0%gcc@6.4.0 build_type=RelWithDebInfo arch=darwin-sierra-x86_64
    ^highfive@1.6%gcc@6.4.0~boost build_type=RelWithDebInfo +mpi arch=darwin-sierra-x86_64
        ^hdf5@1.10.3%gcc@6.4.0~cxx~debug~fortran~hl+mpi+pic+shared~szip~threadsafe arch=darwin-sierra-x86_64
            ^hpe-mpi@2.16%gcc@6.4.0 arch=darwin-sierra-x86_64
            ^zlib@1.2.11%gcc@6.4.0+optimize+pic+shared arch=darwin-sierra-x86_64
    ^py-pybind11@2.2%gcc@6.4.0 build_type=RelWithDebInfo arch=darwin-sierra-x86_64
        ^py-setuptools@40.2.0%gcc@6.4.0 arch=darwin-sierra-x86_64
            ^python@2.7.15%gcc@6.4.0+dbm~optimizations patches=123082ab3483ded78e86d7c809e98a804b3465b4683c96bd79a2fd799f572244 +pic+pythoncmd~shared~tk~ucs4 arch=darwin-sierra-x86_64
                ^bzip2@1.0.6%gcc@6.4.0+shared arch=darwin-sierra-x86_64
                ^gdbm@1.14.1%gcc@6.4.0 arch=darwin-sierra-x86_64
                    ^readline@7.0%gcc@6.4.0 arch=darwin-sierra-x86_64
                ^pkg-config@0.27.1%gcc@6.4.0+internal_glib patches=49ffcd644e190dc5efcb2fab491177811ea746c1a526f75d77118c2706574358 arch=darwin-sierra-x86_64
                ^sqlite@3.23.1%gcc@6.4.0~functions arch=darwin-sierra-x86_64

```

Could you provide more info? (Or, re-producer spack instance in BB5?)

Reply:
Once again, the thing got sorted out somehow as soon as I posted it here, without me making any changes; and I can't reproduce it anymore (was 100% previously).
I'm starting to get afraid for my sanity. ^^

Anyway, so far `+python` is not used by any client component.
If I manage to reproduce the issue, we can address it in a separate PR.

Reply:
Only reason I could imagine is `.pyc` files. Next time try removing all pyc from spack repo and let us know if you still see weird issue. In **addition**, spack maintains some index cache that you can remove with:

```
spack clean -a
```

Reply:
OK, thanks for the tip.